### PR TITLE
[CHAT-2294] GetTask endpoint to retrieve async task results

### PR DESCRIPTION
**Pull request by gumuz**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Changelog

-

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -3279,4 +3280,15 @@ export class StreamChat<
...
+
+  /**
+   * getTask - Gets status of a long running task
+   *
+   * @param {string} id Task ID
+   *
+   * @return {TaskStatus} The task status
+   */
+  async getTask(id: string) {
+    return this.get<TaskStatus>(`${this.baseURL}/tasks/${id}`);
```
Comment (ferhatelmas) [inline src/client.ts:3292]: add `APIResponse` too

Comment (gumuz) [reply src/client.ts:3292]:
```suggestion
    return this.get<APIResponse & TaskStatus>(`${this.baseURL}/tasks/${id}`);
```

File: src/types.ts [hunk 2]
```diff
@@ -2146,3 +2146,11 @@ export type Campaign = {
   updated_at: string;
 } & CampaignData &
   CampaignStatus;
+
+export type TaskStatus = {
+  created_at: string;
+  status: string;
+  task_id: string;
+  updated_at: string;
+  result?: object;
```
Comment (ferhatelmas) [inline src/types.ts:2155]: `error` is missing

File: src/types.ts [hunk 3]
```diff
@@ -2146,3 +2146,15 @@ export type Campaign = {
...
   CampaignStatus;
+
+export type TaskStatus = {
+  created_at: string;
+  status: string;
+  task_id: string;
+  updated_at: string;
+  error?: {
+    description: string;
+    type: string;
```
Comment (ferhatelmas) [inline src/types.ts:2157]: stacktrace and version?

Comment (gumuz) [reply src/types.ts:2157]: there are omitted to to prevent leaking internal info

Comment (ferhatelmas) [reply src/types.ts:2157]: good

**Review by ferhatelmas — DISMISSED** [review]
LGTM

File: src/types.ts [hunk 4]
```diff
@@ -2146,3 +2146,15 @@ export type Campaign = {
...
+  };
+  result?: object;
```
Comment (mahboubii) [inline src/types.ts:2159]: could this get expanded a bit more?

Comment (gumuz) [reply src/types.ts:2159]: so this result in of an undefined type/shape, what do you suggest the typedef should be in that case?

Comment (mahboubii) [reply src/types.ts:2159]: it's either `undefined` or an unknown object? in this case:
```suggestion
  result?: UnknownType;
```

**Review by mahboubii — APPROVED** [review]

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**